src/AccessibilityInsights/Modes/TestModeControl.xaml.cs

```diff
@@ -161,7 +162,10 @@ public async Task SetElement(Guid ecId)
                                 dc.ElementCounter.UpperBound);
                             IsSaveEnabled = false;
                         }
-                        dc.PublishScanResults();
+                        if (publishScanResults)
```

Comment:
`CaptureAction.SetTestModeDataContext` returns a bool that tells us whether the DataContext was updated (in the case, whether or not a scan was run). We also already have `DataContextMode`, which tells us if a file was loaded. We could leverage these two values and avoid manually tracking `PublishScanResultsOneShot`

Reply:
Thanks--that worked!